src/main/java/io/nats/client/impl/NatsJetStream.java

```diff
@@ -419,15 +419,19 @@ public boolean wouldBeChangeTo(ConsumerConfiguration serverCc) {
                 || (flowControl != null && flowControl != serverCcc.isFlowControl())
                 || (headersOnly != null && headersOnly != serverCcc.isHeadersOnly())
 
-                || CcNumeric.START_SEQ.wouldBeChange(startSeq, serverCcc.startSeq)
-                || CcNumeric.MAX_DELIVER.wouldBeChange(maxDeliver, serverCcc.maxDeliver)
-                || CcNumeric.RATE_LIMIT.wouldBeChange(rateLimit, serverCcc.rateLimit)
-                || CcNumeric.MAX_ACK_PENDING.wouldBeChange(maxAckPending, serverCcc.maxAckPending)
-                || CcNumeric.MAX_PULL_WAITING.wouldBeChange(maxPullWaiting, serverCcc.maxPullWaiting)
+                || CcChangeHelper.START_SEQ.wouldBeChange(startSeq, serverCcc.startSeq)
```

Comment:
Why aren't the values simple pass-throughs? Is this because when you serialize you may be resetting values the user set?

Reply:
because usually a user setting any value would be a "change" but in the case of the user sets the same value as the server default, it wouldn't be considered a change. So wouldBeChange checks for unset user value, set user value to something like -1 which means server default and an actual value. It's because my builder is too flexible a straight compare is no good.